txnkv/transaction/2pc.go

```diff
@@ -249,6 +265,8 @@ type PlainMutations struct {
 	keys              [][]byte
 	values            [][]byte
 	isPessimisticLock []bool
+	isAssertExist     []bool
+	isAssertNotExist  []bool
```

Comment:
It looks a bit wasteful to me... Maybe also define "flags" here as the lowest 3 bits of `UserData`?